Update tutorial with recent syntax changes (#7274)
roc-lang/roc: www/content/tutorial.md

`Result.try` is often used to chain two functions that return `Result` (as in the example above). This prevents you from needing to add error handling code at every intermediate step.

### [The `try` keyword](#the-try-keyword) {#the-try-keyword}

Roc also has a special `try` keyword, which is convenient syntax sugar for `Result.try`. For example, consider the following `getLetter` function:

```roc
getLetter : Str -> Result Str [OutOfBounds, InvalidNumStr]
getLetter = \indexStr ->
    index = try Str.toU64 indexStr
    List.get ["a", "b", "c", "d"] index
```

Here's what this does:

> - If the `Str.toU64` function returns an `Ok` value, then its payload is unwrapped.

Reviewer: ```suggestion
- If the `Str.toU64` function returns an `Ok` value, then `try` will return what's inside the `Ok`. For example:
```
"its payload is unwrapped" may be quite complicated for non-native English speakers. I also like the "For example" on that line, that makes it clear you don't have to deeply think about what the sentence is saying because we're giving an example.

Author: Nice!

>   - For example, if we call `getLetter "2"`, then `Str.toU64` returns `Ok 2`, and the `try` keyword unwraps the integer 2, so `index` is set to 2 (not `Ok 2`). Then the `List.get` function is called and returns `Ok "c"`.

Reviewer: ```suggestion
  - If we call `getLetter "2"`, then `Str.toU64` returns `Ok 2`, and the `try` unwraps to the integer 2, so `index` is set to 2 (not `Ok 2`). Then the `List.get` function is called and returns `Ok "c"`.
```

- If the `Str.toU64` function returns an `Err` value, then the `try` keyword immediately interrupts the `getLetter` function and makes it return this error.
  - For example, if we call `getLetter "abc"`, then the call to `Str.toU64` returns `Err InvalidNumStr`, and the `try` keyword ensures that the `getLetter` function returns this error immediately, without executing the rest of the function.

Thanks to the `try` keyword, your code can focus on the "happy path" (where nothing fails) and simply bubble up to the caller any error that might occur. Your error handling code can be neatly separated, and you can rest assured that you won't forget to handle any errors, since the compiler will let you know. See this [code example](https://github.com/roc-lang/examples/blob/main/examples/Results/main.roc) for more details on error handling.

### [The `?` postfix "try" operator](#the-postfix-try-operator) {#the-postfix-try-operator}

STATUS: Deprecated! For now, this may only work within non-PI contexts.

Note: Right now, this can lead to confusing and unhelpful error messages.

> Roc also has a `?` postfix operator, which is like a version of the `try` keyword for `Task`s:

Reviewer: `?` is also for `Result` not for `Task`
```suggestion
Roc also has a `?` postfix operator, which behaves similar to the `try` keyword. An example:
```

Reviewer: Oh I get the confusion now, let me change my suggestion

```roc
getLetter : Str -> Result Str [OutOfBounds, InvalidNumStr]
getLetter = \indexStr ->
    index = Str.toU64? indexStr
    List.get ["a", "b", "c", "d"] index
```

> 

Reviewer: ```suggestion

If you're working with [purity inference](#purity-inference) you'll want to use `try` and if your code is using `Task` you'll want to use `?.`
```
